cmd/crank/beta/migrate/deploymentruntime/cmd.go

```diff
@@ -0,0 +1,98 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package deploymentruntime contains the logic for converting a
+// ControllerConfig to a DeploymentRuntimeConfig.
+package deploymentruntime
+
+import (
+	"github.com/spf13/afero"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	"github.com/crossplane/crossplane-runtime/pkg/errors"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+
+	"github.com/crossplane/crossplane/apis/pkg/v1alpha1"
+	"github.com/crossplane/crossplane/apis/pkg/v1beta1"
+	"github.com/crossplane/crossplane/cmd/crank/beta/migrate/io"
+)
+
+// Cmd arguments and flags for migrate deployment-runtime subcommand.
+type Cmd struct {
+	// Arguments.
+	InputFile  string `short:"i" type:"path" placeholder:"PATH" help:"The ControllerConfig file to be Converted. If not specified, stdin will be used."`
+	OutputFile string `short:"o" type:"path" placeholder:"PATH" help:"The file to write the generated DeploymentRuntimeConfig to. If not specified, stdout will be used."`
+
+	fs afero.Fs
+}
+
+// Help returns help message for the migrate deployment-runtime command.
+func (c *Cmd) Help() string {
+	return `
+This command converts a Crossplane ControllerConfig to a DeploymentRuntimeConfig.
+
+DeploymentRuntimeConfig was introduced in Crossplane 1.14 and ControllerConfig is
+deprecated.
+
+Examples:
+
+  # Write out a DeploymentRuntimeConfigFile from a ControllerConfig
+  crossplane beta migrate deployment-runtime -i cc.yaml -o drc.yaml
+
+  # Create a new DeploymentRuntimeConfigFile via Stdout
+  crossplane beta migrate deployment-runtime -i cc.yaml | grep -v creationTimestamp | kubectl apply -f - 
+
+`
+}
+
+// AfterApply implements kong.AfterApply.
+func (c *Cmd) AfterApply() error {
+	c.fs = afero.NewOsFs()
+	return nil
+}
+
+// Run converts a ControllerConfig to a DeploymentRuntimeConfig.
+func (c *Cmd) Run(logger logging.Logger) error {
+	data, err := io.Read(c.fs, c.InputFile)
+	if err != nil {
+		return err
+	}
+
+	// Set up schemes for our API types
+	sch := runtime.NewScheme()
+	_ = scheme.AddToScheme(sch)
+	_ = v1alpha1.AddToScheme(sch)
+	_ = v1beta1.AddToScheme(sch)
+
+	decode := serializer.NewCodecFactory(sch).UniversalDeserializer().Decode
+
+	cc := &v1alpha1.ControllerConfig{}
+	_, _, err = decode(data, &v1alpha1.ControllerConfigGroupVersionKind, cc)
+	if err != nil {
+		return errors.Wrap(err, "Decode Error")
+	}
+	if cc.Spec.ServiceAccountName != nil && *cc.Spec.ServiceAccountName != "" {
+		logger.Info("WARNING: serviceAccountName is set in the ControllerConfig.\nDeploymentRuntime does not create serviceAccounts, please create the service account separately.", "serviceAccountName", *cc.Spec.ServiceAccountName)
```

Comment:
I believe we can achieve the same behavior by setting [metadata.name for the serviceAccount](https://docs.crossplane.io/v1.14/api/#DeploymentRuntimeConfig-spec-serviceAccountTemplate-metadata-name) just like the example [here](https://docs.crossplane.io/v1.14/concepts/providers/#configuring-metadata-of-runtime-resources).

Reply:
Actually we are setting it [here](https://github.com/crossplane/crossplane/pull/5275/files#diff-5d3fea2ac7d52a09d4c271f82b5a18b1067a951120e040cf5952c3765e2fccdeR198), so I guess we are good and this warning is not needed

Reply:
yes, I think it can be removed too.